docs/subnets/create-a-simple-rust-vm.md

```diff
@@ -0,0 +1,1256 @@
+        Self { vm }
+    }
+}
+
+impl Rpc for Service {
+    fn ping(&self) -> BoxFuture<Result<crate::api::PingResponse>> {
+        log::debug!("ping called");
+        Box::pin(async move { Ok(crate::api::PingResponse { success: true }) })
+    }
+
+    fn propose_block(&self, args: ProposeBlockArgs) -> BoxFuture<Result<ProposeBlockResponse>> {
+        log::debug!("propose_block called");
+        let vm = self.vm.clone();
+
+        Box::pin(async move {
+            vm.propose_block(args.data)
+                .await
+                .map_err(create_jsonrpc_error)?;
+            Ok(ProposeBlockResponse { success: true })
+        })
+    }
+
+    fn last_accepted(&self) -> BoxFuture<Result<LastAcceptedResponse>> {
+        log::debug!("last accepted method called");
+        let vm = self.vm.clone();
+
+        Box::pin(async move {
+            let vm_state = vm.state.read().await;
+            if let Some(state) = &vm_state.state {
+                let last_accepted = state
+                    .get_last_accepted_block_id()
+                    .await
+                    .map_err(create_jsonrpc_error)?;
+
+                return Ok(LastAcceptedResponse { id: last_accepted });
+            }
+
+            Err(Error {
+                code: ErrorCode::InternalError,
+                message: String::from("no state manager found"),
+                data: None,
+            })
+        })
+    }
+
+    fn get_block(&self, args: GetBlockArgs) -> BoxFuture<Result<GetBlockResponse>> {
+        let blk_id = ids::Id::from_str(&args.id).unwrap();
+        log::info!("get_block called for {}", blk_id);
+
+        let vm = self.vm.clone();
+
+        Box::pin(async move {
+            let vm_state = vm.state.read().await;
+            if let Some(state) = &vm_state.state {
+                let block = state
+                    .get_block(&blk_id)
+                    .await
+                    .map_err(create_jsonrpc_error)?;
+
+                return Ok(GetBlockResponse { block });
+            }
+
+            Err(Error {
+                code: ErrorCode::InternalError,
+                message: String::from("no state manager found"),
+                data: None,
+            })
+        })
+    }
+}
+```
+
+In the below examples "2wb1UXxAstB8ywwv4rU2rFCjLgXnhT44hbLPbwpQoGvFb2wRR7" is the blockchain ID.
+
+#### timestampvm.getBlock
+
+```sh
+curl -X POST --data '{
+    "jsonrpc": "2.0",
+    "id"     : 1,
+    "method" : "timestampvm.getBlock",
+    "params" : [{"id":"SDfFUzkdzWZbJ6YMysPPNEF5dWLp9q35mEMaLa8Ha2w9aMKoC"}]
+}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/2wb1UXxAstB8ywwv4rU2rFCjLgXnhT44hbLPbwpQoGvFb2wRR7/rpc
+
+# example response
+# {"jsonrpc":"2.0","result":{"block":{"data":"0x32596655705939524358","height":0,"parent_id":"11111111111111111111111111111111LpoYY","timestamp":0}},"id":1}
+```
+
+#### timestampvm.proposeBlock
+
+```sh
+# to propose data
+echo 1 | base64 | tr -d \\n
+# MQo=
```

Comment:
are lines 1105-1108 needed? Please remove if not.

Reply:
Yeah its just a shell helper which shows how to base64 encode data.